feat(networking): remove NodeIssue::Connection

joshuef opened the pull request.

jacderida commented:
Hey Josh, do the tests not need updated after this change?

jacderida commented on sn_testnet/src/lib.rs:
```diff
@@ -27,15 +27,27 @@ const TESTNET_DIR_NAME: &str = "local-test-network";
+    fn launch(
+        &self,
+        node_name: String,
+        // node_data_path: PathBuf,
```
Just a small one: could we remove that commented out code please?

jacderida commented on sn_testnet/src/lib.rs:
```diff
@@ -265,14 +282,23 @@ impl Testnet {
+
+            std::fs::create_dir_all(node_data_dir_path.clone())?;
+
+            info!("Launch args are: {launch_args:?}");
```
As above.

joshuef committed:
feat(networking): remove NodeIssue::Connection

Temmporary issues might cause increased shunning here. In reality any connection issues
should surface through replication and data verificaiton tests (or they are not
testing sufficiently).

So this removes a potential source of failure of nodes/networks due to
temporary issues
